x/posts/legacy/v3/store.go

```diff
@@ -153,3 +174,127 @@ func migrateReplySettings(settings v2.ReplySetting) types.ReplySetting {
 		panic(fmt.Errorf("invalid reply settings value: %s", settings))
 	}
 }
+
+// migrateAttachments migrates the attachments present inside the store from v2 to v3
+func migrateAttachments(store sdk.KVStore, cdc codec.BinaryCodec) error {
+	prefixStore := prefix.NewStore(store, types.AttachmentPrefix)
+	iterator := prefixStore.Iterator(nil, nil)
+
+	// Get all the attachments
+	var v2Attachments []v2.Attachment
+	for ; iterator.Valid(); iterator.Next() {
+		var v2Attachment v2.Attachment
+		cdc.MustUnmarshal(iterator.Value(), &v2Attachment)
+		v2Attachments = append(v2Attachments, v2Attachment)
+	}
+
+	// Close the iterator
+	err := iterator.Close()
+	if err != nil {
+		return err
+	}
+
+	// Convert the attachments
+	v3Attachments := convertAttachments(v2Attachments)
+	for i, v3Attachment := range v3Attachments {
+		// Save the attachment
+		store.Set(
+			types.AttachmentStoreKey(v3Attachment.SubspaceID, v3Attachment.PostID, v3Attachment.ID),
+			cdc.MustMarshal(&v3Attachments[i]),
```

Comment:
It can be `cdc.MustMarshal(&v3Attachment)` then remove `i` from the for loop.

Reply:
No, as using that code would trigger this lint error (as it can be seen [here](https://github.com/desmos-labs/desmos/runs/7147444000?check_suite_focus=true)): 
```
G601: Implicit memory aliasing in for loop
```